docs/tutorials/enable-public-entry.md

```diff
@@ -0,0 +1,118 @@
+
+- Public entry point containing login pages;
+- There is also a protected main entry point that contains the code for what you see after signing in.
+
+With that, Backstage's cli and backend will detect public entry point and serve it to unauthenticated users, while serving the main, protected entry point only to authenticated users.
+
+## Requirements
+
+- The tutorial will only work for those using backstage-cli to build and serve their Backstage app.
+
+## Trying out this feature
+
+1. Add a `index-public-experimental.tsx` to your app `src` folder;
+
+2. Prepare an unauthenticated version of your application. This will be the public entry point for your site:
+
+   ```tsx title="in packages/app/src/index-public-experimental.tsx"
+   import React from 'react';
+   import ReactDOM from 'react-dom/client';
+   import { createApp } from '@backstage/app-defaults';
+   import { AppRouter } from '@backstage/core-app-api';
+   import {
+     AlertDisplay,
+     OAuthRequestDialog,
+     SignInPage,
+   } from '@backstage/core-components';
+   import {
+     configApiRef,
+     createApiFactory,
+     discoveryApiRef,
+   } from '@backstage/core-plugin-api';
+   import { RedirectToRoot } from '@backstage/plugin-auth-react';
+   import { providers } from '../src/identityProviders';
+   import { AuthProxyDiscoveryApi } from '../src/AuthProxyDiscoveryApi';
+
+   // Notice that this is only setting up what is needed by the sign-in pages
+   const app = createApp({
+     apis: [
+       createApiFactory({
+         api: discoveryApiRef,
+         deps: { configApi: configApiRef },
+         factory: ({ configApi }) =>
+           AuthProxyDiscoveryApi.fromConfig(configApi),
+       }),
+     ],
+     components: {
+       SignInPage: props => {
+         return (
+           <SignInPage
+             {...props}
+             providers={['guest', 'custom', ...providers]}
+             title="Select a sign-in method"
+             align="center"
+           />
+         );
+       },
+     },
+   });
+
+   const App = app.createRoot(
+     <>
+       <AlertDisplay transientTimeoutMs={2500} />
+       <OAuthRequestDialog />
+       <AppRouter>
+         {/* This is a special component that does the magic to redirect users to access the home page of your authenticated application version */}
+         <RedirectToRoot />
+       </AppRouter>
+     </>,
+   );
+
+   ReactDOM.createRoot(document.getElementById('root')!).render(<App />);
+   ```
+
+3) Then ensure your main entry is also protected by the experimental app provider, as shown in the following example:
+
+```tsx title="in packages/app/src/index-public-experimental.tsx"
+// ...
+export default app.createRoot(
+  <>
+    <AlertDisplay transientTimeoutMs={2500} />
+    <OAuthRequestDialog />
+    <AppRouter>
+      {/* For now, this is just a temporary solution to ensure the user remains logged in properly and has access to the main app content. */}
+      <ExperimentalAppProtection>
```

Comment:
This seems unnecessary to do manually, I would have placed it in the createRoot logic, but we can't have a dependency on the `auth-react` package from the `core-plugin-api` package, can we?

Reply:
I realized that the cookie refresh provider expects a children, so it needs to be a wrap because of that.